src/Keboola/StorageApi/Downloader/DownloaderFactory.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Keboola\StorageApi\Downloader;
+
+use Aws\S3\S3Client;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\Exception;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+
+class DownloaderFactory
+{
+    /**
+     * @param $fileInfo
+     * @param int $retries
+     * @return DownloaderInterface
+     * @throws Exception
+     */
+    public static function createDownloaderForPrepareRequest($fileInfo, $retries = Client::DEFAULT_RETRIES_COUNT)
```

Comment:
Co je tam to "PrepareRequest"?

Reply:
Dovnitř do toho padá $fileInfo z KBC a počet retries. Tak bych čekal, že to bude "createDownloaderForExportedKbcFile()" nebo tak něco

Reply:
Nebo bych čekal, že se bude jinak jmenovat ta proměnná...

Reply:
$fileInfo je z `v2/storage/files/prepare` proto je tam ForPrepareRequest, zanechal sem původní názvy proměnných. Možná je se to může jmenovat $fileInfoResponse  nebo $prepareRequestResponse nebo ta metoda može byt createDownloaderForPrepareRequestResponse :)

Reply:
Je? Podle mého ne. Tady se tahá:
https://github.com/keboola/storage-api-php-client/pull/361/files#diff-88c691f964f19f54df0a02bd42e20b8aR42-R45

A tady je implementace toho getFile https://github.com/keboola/storage-api-php-client/blob/c79296dc970612f14af2c1c9ea76b9c9ea76af6b/src/Keboola/StorageApi/Client.php#L873

Nebo koukám špatně?

Reply:
Jop sorry já sem sa špatně díval https://github.com/keboola/storage-api-php-client/pull/361/commits/c0e2f7b1878b5cc494720bec20a13233741c3220

Reply:
Klidně bych teda nechal jen createDownloader... ale i takhle OK